lib/base-compiler.js

```diff
@@ -1675,14 +1695,16 @@ export class BaseCompiler {
         }
 
         if (!backendOptions.skipAsm) {
-            if (result.okToCache) {
-                const res = this.processAsm(result, filters, options);
-                result.asm = res.asm;
-                result.labelDefinitions = res.labelDefinitions;
-                result.parsingTime = res.parsingTime;
-                result.filteredCount = res.filteredCount;
-            } else {
-                result.asm = [{text: result.asm}];
+            if (typeof result.asm === 'string') {
```

Comment:
I don't quite understand what this is doing here - is this handling the situation where the asm has already been decoded? It seems quite subtle to use the `result.asm` object's type in this way. (presumably if it's not a string, the implication is it's a dictionary that was created by something above?

Reply:
I wish there was a better place to do this, it's not pretty. (The naming of the okToCache check here is also incredibly confusing)

But yes, `.asm` is an object when it returns from `objdump` where the result is fed through the external parser, so we need to check here that we don't need to feed it through `processAsm` again.

We could do the check in processAsm, but that's sometimes overridden and then we need to keep that up-to-date.

Reply:
Maybe we can give it a better name like `result.alreadyParsed` and then skip the parsing step